pkg/providers/imagefamily/bootstrap/aksbootstrap.go

```diff
@@ -283,40 +221,74 @@ func kubeBinaryURL(kubernetesVersion, cpuArch string) string {
 	return fmt.Sprintf("%s/kubernetes/v%s/binaries/kubernetes-node-linux-%s.tar.gz", globalAKSMirror, kubernetesVersion, cpuArch)
 }
 
-func (a AKS) applyOptions(nbv *nbcontractv1.Configuration) {
-	nbv.ClusterCertificateAuthority = *a.CABundle
-	nbv.ApiserverConfig.ApiServerName = a.APIServerName
-	nbv.TlsBootstrappingConfig.TlsBootstrapToken = a.KubeletClientTLSBootstrapToken
+// Check and initialize each field if it is nil
+func initializeIfNil[T any](field **T) {
+	if *field == nil {
+		*field = new(T)
+	}
+}
+
+// EnsureConfigsNonNil checks if the config is nil and initializes it if it is.
+func ensureConfigsNonNil(v *nbcontractv1.Configuration) *nbcontractv1.Configuration {
+	if v == nil {
+		v = &nbcontractv1.Configuration{}
+	}
+
+	initializeIfNil(&v.KubeBinaryConfig)
+	initializeIfNil(&v.ApiServerConfig)
```

Comment:
instead of initializing all the structs here, im still wondering if it makes more sense to initialize the structs as we know we are using them. e.g. right before we set `ApiServerConfig.ApiServerName` in `applyOptions` we call `initializeIfNil(&v.ApiServerConfig)`, etc. That way we can more easily keep track of whether the structs are initialized before they are used instead of having a long list here.